### PR TITLE
enable redundant preprocessor directive check in travis

### PR DESCRIPTION
**Pull request by FranciscoPombal**
Just to see what happens. Noticed while looking at `http_connection.cpp` (`#if TORRENT_USE_I2P`).

### CODE REVIEW

**Comment by FranciscoPombal** [comment]
@arvidn How can I actually see the warnings in https://travis-ci.org/github/arvidn/libtorrent/jobs/701706030?

>  ```
> clang_tidy.compile.c++ bin/clang_tidy-7.0.0/debug/crypto-openssl/cxxstd-17-iso/src/alert_manager.o
> 544 warnings generated.
> ...
> ```

I'm mainly interested in seeing if any `readability-redundant-preprocessor` came up.

**Comment by arvidn** [comment]
I need to fix the patch. I was trying to avoid the problem by using debug iterators (since they won’t be pointers) but that causes the link errors instead

**Comment by FranciscoPombal** [comment]
@arvidn 

> I need to fix the patch. I was trying to avoid the problem by using debug iterators (since they won’t be pointers) but that causes the link errors instead

Misread PR title? :)
The linker errors occurred with https://github.com/arvidn/libtorrent/pull/4789, not with this one.

**Comment by arvidn** [comment]
heh, yes :) I blame being on the phone

**Comment by arvidn** [comment]
If there had been any hits with that checker, `clang_tidy` would have failed and caused the CI to fail. So I'm pretty sure this means there are no hits with that checker

**Comment by FranciscoPombal** [comment]
@arvidn 

> If there had been any hits with that checker, `clang_tidy` would have failed and caused the CI to fail. So I'm pretty sure this means there are no hits with that checker

That's strange, there should have been at least one hit:

https://github.com/arvidn/libtorrent/blob/db85195c8c5a7eb3cf1d484f68048aaec4c4f22d/src/http_connection.cpp#L272
https://github.com/arvidn/libtorrent/blob/db85195c8c5a7eb3cf1d484f68048aaec4c4f22d/src/http_connection.cpp#L286

I get that hit in my machine:

```bash
$ clang-tidy-11 -checks="-*,readability-redundant-preprocessor" src/http_connection.cpp                  
1 warning generated.
/home/francisco/Documents/libtorrent/src/http_connection.cpp:290:2: warning: nested redundant #if; consider removing it [readability-redundant-preprocessor]
#if TORRENT_USE_I2P
 ^
/home/francisco/Documents/libtorrent/src/http_connection.cpp:276:2: note: previous #if was here
#if TORRENT_USE_I2P
 ^
```

I also get that hit in my editor with `clangd` assistance (`clangd  -j=8 --clang-tidy --all-scopes-completion --background-index --completion-style=detailed --header-insertion=iwyu --suggest-missing-includes`) + a `.clang-tidy` file for enabling the appropriate checks.

Then again, I'm using `clang-tidy`/`clangd` 11 for this; perhaps the version in the CI is too low?

**Comment by arvidn** [comment]
> Then again, I'm using clang-tidy/clangd 11 for this; perhaps the version in the CI is too low?

Yeah, travis has clang 7 [[link](https://travis-ci.org/github/arvidn/libtorrent/jobs/701706030#L432)]

**Comment by FranciscoPombal** [comment]
@arvidn You can check all supported checks with `clang-tidy -checks="*" --explain-config`

Looks like for this one, at least `clang-tidy` 8 is required. For `readability-qualified-auto` in https://github.com/arvidn/libtorrent/pull/4789, the minimum requirement is version 10.

**Comment by arvidn** [comment]
clang-tidy-7 is running on travis now. you may want to rebase this

**Comment by FranciscoPombal** [comment]
@arvidn 

got 'em:
https://travis-ci.org/github/arvidn/libtorrent/jobs/704277263#L771
https://travis-ci.org/github/arvidn/libtorrent/jobs/704277263#L850
https://travis-ci.org/github/arvidn/libtorrent/jobs/704277263#L1058
https://travis-ci.org/github/arvidn/libtorrent/jobs/704277263#L1075

I'll introduce a second commit in this PR fixing these, is that ok?

**Comment by arvidn** [comment]
Sure

**Comment by FranciscoPombal** [comment]
@arvidn https://github.com/arvidn/try_signal/pull/16

**Comment by FranciscoPombal** [comment]
@arvidn https://travis-ci.org/github/arvidn/libtorrent/builds/704811249

**Comment by FranciscoPombal** [comment]
@arvidn appveyor build failed with 

```
00:00:03.177: ses1: [torrent_log] temporary: init, async_check_files
0.0s | 0kB/s 0kB/s 100% 0 cc:1 : 0.0s | 0kB/s 0kB/s 0% 0 cc:0
st1-progress: 1 7
st2-progress: 0 3
***** test_transfer.cpp:257 "TEST_ERROR: check failed: "st1.state == torrent_status::seeding || st1.state == torrent_status::checking_files"" *****
st1-progress: 1 7
st2-progress: 0 3
***** test_transfer.cpp:257 "TEST_ERROR: check failed: "st1.state == torrent_status::seeding || st1.state == torrent_status::checking_files"" *****
```

Spurious error?

**Comment by arvidn** [comment]
yes, this test has started to fail sometimes on appveyor.. someone should do something about it :)

**Comment by FranciscoPombal** [comment]
@arvidn 

> yes, this test has started to fail sometimes on appveyor.. someone should do something about it :)

Well, I force-pushed (simple no-change rebase) to re-run the CI just as you posted this, sorry about that. Anyway, the PR is ready from my side.

File: src/session_impl.cpp [hunk 1]
```diff
@@ -805,9 +803,7 @@ bool ssl_server_name_callback(ssl::stream_handle_type stream_handle, std::string
 			}
 		}
 
-#if TORRENT_ABI_VERSION <= 2
```
Comment (arvidn) [inline src/session_impl.cpp:808]: I don't see how this can be redundant, if the one on line 802 isn't..

Comment (arvidn) [reply src/session_impl.cpp:808]: I see, line 677

**Pull request merged**